### ISSUE TITLE
Starting leads to immediate crash

### ISSUE THREAD

**Issue by dyfustic** [feditext/feditext#344]
**Describe the bug**
Since yesterday Feditext crashes and is not able to ever get back to running. It started with having all navigation available but the timeline being empty. Touching the navigation items did nothing, the app was frozen. Stopping the app and starting it again produced an error message “Feditext has crashed”. I even restarted the phone. Same result. The app is currently unusable.

**To Reproduce**
1. Go to Feditext App icon 
2. Click on it
3. Wait a little bit and watch the loading symbol move. 
4. See error message show up on screen 

**Expected behavior**
The app should start up and run.

**Screenshots**
![IMG_2944](https://github.com/feditext/feditext/assets/94124819/3659c235-ebec-4ac7-92fd-9c3089871797)
![IMG_2943](https://github.com/feditext/feditext/assets/94124819/2d593da6-38a7-4835-b4ec-0b0527a19add)

**(please complete the following information):**
 - Device: iPhone 15 pro
 - OS: iOS 17.2.1
 - App Version 1.7.2 (80)

**Additional context**
Now to the odd thing: I am using Metatext on the same device. And it was showing the exact same behaviour starting at the same time. Deleting and re-installing it fixed the issue here. Hardware issue? Same library used in both apps? 

Update: 1.7.2 (83) got installed overnight. Still crashes. Now I have to figure out, how to reinstall an app in TestFlight or better figure out where I left the access info for the beta test. 🤷🏼‍♀️

**Comment by LuNeder** [comment]
Exact same problem here, but started today in the morning instead 

iOS 17.0  
iPhone 14 Pro  
1.7.2 (80) (also happens on 79, 78 and Metatext)

On Metatext I don’t have any accounts logged in, but this bug happens as soon as I type my instances url and press login. The loading spinny wheel shows up and stays permanently there - no safari popup is ever opened.

**Comment by LuNeder** [comment]
> Exact same problem here, but started today in the morning instead
> 
> iOS 17.0 iPhone 14 Pro 1.7.2 (80) (also happens on 79, 78 and Metatext)
> 
> On Metatext I don’t have any accounts logged in, but this bug happens as soon as I type my instances url and press login. The loading spinny wheel shows up and stays permanently there - no safari popup is ever opened.

I sent the report and logs when prompted by the TestFlight abend/crash popup, hopefully that’ll have something useful in the logs

Edit: Can confirm that uninstalling Feditext and installing it again fixes the issue

**Comment by VyrCossont** [comment]
Thanks for the reports! Hopefully we can get something useful out of the crash logs.

**Comment by vy-let** [comment]
I'm seeing this same issue — it's happened about three or four times now, starting sometime last fall, and the only resolution I've been able to find is deleting and reinstalling the app. (Noooo, my settings! 😭) Sometimes it crashes in a way that I can submit a report, but a lot of times (including right now) it just hangs at launch. If I leave it open in this hung state, my phone gets hot.

There did not seem to be any incident that caused this. In each case, it was preceded by a quick ramp-up in slowness in the app.

If there's anything I can do to help track down the problem before reinstalling again, I'd like to try.

N.B. I wasn't sure whether this is the same issue as #255 — I'm posting here because I didn't see a direct correlation with an iOS update.

**Comment by VyrCossont** [comment]
Bumping this up the list.

**Comment by VyrCossont** [comment]
Found a common cause for a handful of crashes either during startup, when adding a new instance, or when refreshing info on an existing one: an out-of-bounds case in the instance version number parser. Could explain #255 and/or this bug.

I need some info from y'all to confirm that this is the issue you're hitting (or not): the software version strings from your instance's `/nodeinfo/2.0` or `/nodeinfo/2.1` or `/nodeinfo/2.0.json` or `/nodeinfo/2.1.json`, whichever URL exists. I don't need to know the instance domain if you don't feel like sharing it, but it might help.

For example, one of my instances has its [NodeInfo](https://nodeinfo.diaspora.software/) document at https://demon.social/nodeinfo/2.0 and returns the following JSON, from which the software version string is `4.3.0-alpha.1+glitch`:

```json
{
  "version": "2.0",
  "software": {
    "name": "mastodon",
    "version": "4.3.0-alpha.1+glitch"
  },
  "protocols": [
    "activitypub"
  ],
  "services": {
    "outbound": [],
    "inbound": []
  },
  "usage": {
    "users": {
      "total": 5,
      "activeMonth": 3,
      "activeHalfyear": 4
    },
    "localPosts": 41443
  },
  "openRegistrations": false,
  "metadata": {
    "nodeName": "demon.social",
    "nodeDescription": "Pocket-sized friends-only instance. Registrations closed."
  }
}
```

**Comment by LuNeder** [comment]
Mine (tech.lgbt) is currently 4.3.0-alpha.1+glitch, but I know that last week it was alpha.0. Not sure what it was when the problem happened to me (I’ve since reinstalled feditext and metatext, which fixed the problem in both) tho.

**Comment by VyrCossont** [comment]
I don't think that can trigger the version bug, so I'll keep working down the crash list. Some of the other top crashes seem likely to be triggered by searches or by emoji reactions on posts, but the first wouldn't cause a crash on startup, and the second won't apply to most Mastodon instances, including `tech.lgbt`.

**Comment by weirdofhermiston** [comment]
Just had this issue happen again. 

On toot.cat it's `{"version":"2.0","software":{"name":"mastodon","version":"4.3.0-alpha.1+glitch"},"`

**Comment by VyrCossont** [comment]
While severe, we need more crash dumps to have a hope of catching this, so I'm bumping it to the next milestone.
